mmcv/runner/base_module.py

```diff
@@ -1,31 +1,76 @@
 # Copyright (c) Open-MMLab. All rights reserved.
+import copy
 import warnings
 from abc import ABCMeta
+from collections import defaultdict
+from logging import FileHandler
 
 import torch.nn as nn
 
-from mmcv import ConfigDict
+from mmcv.runner.dist_utils import master_only
+from mmcv.utils.logging import get_logger, logger_initialized, print_log
+
+
+def update_init_info(module, *, init_info):
+    """Update the `_params_init_info` in the module if the value of parameters
+    are changed.
+
+    Args:
+        module (obj:`nn.Module`): The module of PyTorch with a user-defined
+            attribute `_params_init_info` which records the initialization
+            information.
+        init_info (str): The string that describes the initialization.
+    """
+    for param in module.parameters():
+        mean_value = param.data.mean()
+        if module._params_init_info[param]['tmp_mean_value'] != mean_value:
+            module._params_init_info[param]['init_info'] = init_info
+            module._params_init_info[param]['tmp_mean_value'] = mean_value
 
 
 class BaseModule(nn.Module, metaclass=ABCMeta):
-    """Base module for all modules in openmmlab."""
+    """Base module for all modules in openmmlab.
 
-    def __init__(self, init_cfg=None):
-        """Initialize BaseModule, inherited from `torch.nn.Module`
+    ``BaseModule`` is a wrapper of ``torch.nn.Module`` with additional
+    functionality of parameter initialization. Compared with
+    ``torch.nn.Module``, ``BaseModule`` mainly adds three attributes.
 
-        Args:
-            init_cfg (dict, optional): Initialization config dict.
-        """
+        - ``init_cfg``: the config to control the initialization.
+        - ``_params_init_info``: Used to track the parameter
+            initialization information.
+        - ``init_weights``: The function of parameter
+            initialization and recording initialization
+            information.
+
+    Args:
+        init_cfg (dict, optional): Initialization config dict.
+    """
+
+    def __init__(self, init_cfg=None):
+        """Initialize BaseModule, inherited from `torch.nn.Module`"""
 
         # NOTE init_cfg can be defined in different levels, but init_cfg
         # in low levels has a higher priority.
 
         super(BaseModule, self).__init__()
         # define default value of init_cfg instead of hard code
-        # in init_weight() function
+        # in init_weights() function
         self._is_init = False
+        if init_cfg:
+            init_cfg = copy.deepcopy(init_cfg)
```

Comment:
No need to adopt `if-else` condition. Just using `init_cfg = copy.deepcopy(init_cfg)` is ok.